core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1959,7 +1960,7 @@ public FormValidation doCheckLabel(@AncestorInPath AbstractProject<?,?> project,
                 }
             }
             return FormValidation.okWithMarkup(Messages.AbstractProject_LabelLink(
-                    j.getRootUrl(), Util.escape(l.getName()), l.getUrl(), l.getNodes().size(), l.getClouds().size())
+                    Stapler.getCurrentRequest().getContextPath() + "/", Util.escape(l.getName()), l.getUrl(), l.getNodes().size(), l.getClouds().size())
```

Comment:
spider senses are pining on this one... but I don;t know why...
maybe because the method as is can be used outside of any web request (but I doubt it ever is)  or maybe for some other reason

Reply:
Will not work with reverse proxy I think if it stupidly rewrites the context path.

Reply:
Do you see an other possibility without relying on the rootURL ? In a sense I applied the same logic as in `Functions#getActionUrl`